File: models/src/main/java/ar/edu/itba/paw/models/Province.java
package ar.edu.itba.paw.models;

import javax.persistence.*;

@Entity
@Table(
        name = "system_province",
        indexes = {
                @Index(columnList = "province_id", name = "system_province_province_id_uindex", unique = true),
        }
)
public class Province extends GenericModel<Integer> {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "system_province_province_id_seq")
    @SequenceGenerator(sequenceName = "system_province_province_id_seq", name = "system_province_province_id_seq", allocationSize = 1)
    @Column(name = "province_id")
    private Integer id;
    @Column(name = "name", nullable = false)
    private String name;
    @ManyToOne(fetch = FetchType.EAGER, optional = false
//            ,
//            cascade = {CascadeType.PERSIST, CascadeType.REFRESH, CascadeType.MERGE}
    )
    @JoinColumn(name = "country_id", nullable = false)
    private Country country;

    @Override
    public Integer getId() {
        return this.id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Country getCountry() {
        return this.country;
    }

    public void setCountry(Country country) {
        this.country = country;
    }

    @Override
    protected boolean isSameType(Object o) {
        return o instanceof Province;
    }
}

File: webapp/src/main/vueapp/src/logic/models/Province.ts
import {GenericEntity} from '~/logic/models/utils/GenericEntity';

export class Province extends GenericEntity<Province> {
    private _id: number;
    private _name: string;
    private _countryId: number;

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get name(): string {
        return this._name;
    }

    public set name(value: string) {
        this._name = value;
    }

    public get countryId(): number {
        return this._countryId;
    }

    public set country(value: number) {
        this._countryId = value;
    }
}
